packages/apps/kadena-docs/src/utils/staticGeneration/checkSubTreeForActive.js

```diff
@@ -0,0 +1,57 @@
+import { getData } from './getData.js';
+import path from 'path';
+
+const isIndex = (filename) => {
+  return filename === 'index';
+};
+
+// we dont want the last option in the path be "index"
+const lastInPath = (filename) => {
+  if (isIndex(filename)) return '';
+  return `/${filename}`;
+};
+
+export const getPathName = () => {
+  const endPoint = 'docs';
+
+  const dirArray = __dirname.split('/');
+  const newPath = dirArray
+    .reverse()
+    .slice(0, dirArray.indexOf(endPoint) + 1)
+    .reverse()
+    .join('/');
+
+  return `/${newPath}${lastInPath(path.parse(__filename).name)}`;
+};
+
+const mapSubTree = (pathname) => (item) => {
+  const newItem = { ...item };
+  // is the menu open?
```

Comment:
since you are asking the question via comments, I'd rather have this if statement extracted as a function asking said question

Reply:
and I'd also return the new item like: 

```
(item) => ({
  ...item,
  isMenuOpen: isMenuOpen(pathName, newItem.root),
  isActive: isActive(pathName, newItem.root),
  children: item.children || newItem.children.map(mapSubTree(pathname))
})
```

Reply:
makes sense